File: DataMonitoring/ClientApp/src/app/widget/models/indicator-bar-widget.ts
import {IndicatorWidget} from "./widget";

export class IndicatorBarWidget extends IndicatorWidget {
    // Axe X
    public displayAxeX = true;

    // Label
    public labelColumnCode = "";  
    public labelFontSize = 20;
    public labelColorText = "#000000";  

    // Axe Y
    public displayAxeY = false;
    public displayDataAxeY = false;
    public textDataAxeYColor = "#000000"; 
    public displayGridLinesAxeY = false;

    // DataLabel
    public dataColumnCode = ""; 
    public displayDataLabelInBar = true;
    public dataLabelInBarColor = "#000000"; 
    public fontSizeDataLabel = 25;
    public decimalMask = "";
    public barColor = "#000000"; 

    // Stacked bar
    public addBarStackedToTarget = true;
    public barColorStackedToTarget = "#000000";  

    // Target
    public addTargetBar = true;
    public targetBarColor = "#000000";
    public setSumDataToTarget = true;

    public indicatorBarWidgetColumnList: IndicatorBarWidgetColumn[]; 
    public barLabelWidgetList: BarLabelWidget[]; 
}

export class IndicatorBarWidgetColumn {
    public id = 0;
    public code = "";
    public filtered = false;
    public filteredValue = "";
    public isNumericFormat = false;
}

export class BarLabelWidget {
    public id = 0;
    public name = "NEW";
    public sequence = 0;
    public labelTextColor = "#000000";
    public useLabelColorForBar = true;

    public barLabelWidgetLocalizationList: BarLabelWidgetLocalization[];
}

export class BarLabelWidgetLocalization {
    public id = 0;
    public localizationCode = "";
    public title = "";
}

File: DataMonitoring.Model/IndicatorBarWidget.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DataMonitoring.Model
{
    public class IndicatorBarWidget : IndicatorWidget
    {
        public bool DisplayAxeX { get; set; }

        public bool DisplayAxeY { get; set; }

        public bool DisplayDataAxeY { get; set; }

        [Required]
        [StringLength(50)]
        public string TextDataAxeYColor { get; set; }

        public bool DisplayGridLinesAxeY { get; set; }

        [Required]
        [StringLength(60)]
        public string LabelColumnCode { get; set; }

        [Required]
        [StringLength(50)]
        public string LabelColorText { get; set; }

        public int LabelFontSize { get; set; }
       
        [Required]
        [StringLength(60)]
        public string DataColumnCode { get; set; }

        public bool DisplayDataLabelInBar { get; set; }

        [Required]
        [StringLength(50)]
        public string DataLabelInBarColor { get; set; }

        public int FontSizeDataLabel { get; set; }

        [StringLength(60)]
        public string DecimalMask { get; set; }
        
        public bool AddTargetBar { get; set; }

        [StringLength(50)]
        public string TargetBarColor { get; set; }

        [StringLength(50)]
        public string BarColor { get; set; }

        public bool SetSumDataToTarget { get; set; }

        public bool AddBarStackedToTarget { get; set; }

        [Required]
        [StringLength(60)]
        public string BarColorStackedToTarget { get; set; }
        
        public ICollection<IndicatorBarWidgetColumn> IndicatorBarWidgetColumns { get; set; }

        public ICollection<BarLabelWidget> BarLabelWidgets { get; set; }
    }
}
